mainwindow.cpp

```diff
@@ -30,4 +31,30 @@ MainWindow::MainWindow(QWidget *parent) :
 MainWindow::~MainWindow()
 {
     delete ui;
+    delete canvas;
+}
+
+void MainWindow::on_actionLoad_triggered()
+{
+    jbrush::FileHandler dataHandler;  // Handles data loading for this method
```

Comment:
Now I don't know how I feel about naming this class ``FileHandler``. It makes it seem like the class is a general class that deals with files. Obviously, it doesn't matter, just an itch.